File: src/main/scala/exoskeleton/common/ExoGuiHandler.scala
package exoskeleton.common

import cpw.mods.fml.common.network.IGuiHandler
import exoskeleton.client.gui._
import exoskeleton.common.gui.{ContainerModifier, ContainerToolbox}
import exoskeleton.common.lib.tree.TreeRecon
import exoskeleton.common.tile.{TileEntityModifier, TileEntityToolbox}
import net.minecraft.entity.player.EntityPlayer
import net.minecraft.tileentity.TileEntity
import net.minecraft.world.World

object ExoGuiHandler
extends IGuiHandler{
  override def getServerGuiElement(ID: Int, player: EntityPlayer, world: World, x: Int, y: Int, z: Int): AnyRef ={
    val tile: TileEntity = world.getTileEntity(x, y, z);
    ID match
    {
      case 0 =>{
        return new ContainerToolbox(tile.asInstanceOf[TileEntityToolbox], player);
      }
      case 1 =>{
        return new ContainerModifier(tile.asInstanceOf[TileEntityModifier], player);
      }
      case _ =>{
        return null;
      }
    }
  }

  override def getClientGuiElement(ID: Int, player: EntityPlayer, world: World, x: Int, y: Int, z: Int): AnyRef ={
    val tile: TileEntity = world.getTileEntity(x, y, z);
    ID match
    {
      case 0 =>{
        return new GuiToolbox(new ContainerToolbox(tile.asInstanceOf[TileEntityToolbox], player));
      }
      case 1=>{
        return new GuiModifier(new ContainerModifier(tile.asInstanceOf[TileEntityModifier], player));
      }
      case 2 =>{
        return new GuiTest(player, TreeRecon);
      }
      case 3 =>{
        return new GuiDebug(player);
      }
      case _ =>{
        return null;
      }
    }
  }
}

File: src/main/scala/exoskeleton/common/EXOGuiHandler.java
package exoskeleton.common;

import cpw.mods.fml.common.network.IGuiHandler;
import exoskeleton.client.gui.GuiDebug;
import exoskeleton.client.gui.GuiModifier;
import exoskeleton.client.gui.GuiTest;
import exoskeleton.client.gui.GuiToolbox;
import exoskeleton.common.gui.ContainerModifier;
import exoskeleton.common.gui.ContainerToolbox;
import exoskeleton.common.lib.tree.TreeRecon;
import exoskeleton.common.tile.TileEntityModifier;
import exoskeleton.common.tile.TileEntityToolbox;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.tileentity.TileEntity;
import net.minecraft.world.World;

public final class EXOGuiHandler
implements IGuiHandler{
    @Override
    public Object getServerGuiElement(int ID, EntityPlayer player, World world, int x, int y, int z){
        TileEntity tile = world.getTileEntity(x, y, z);
        switch(ID)
        {
            case 0:{
                return new ContainerToolbox(((TileEntityToolbox) tile), player);
            }
            case 1:{
                return new ContainerModifier(((TileEntityModifier) tile), player);
            }
            default:{
                return null;
            }
        }
    }

    @Override
    public Object getClientGuiElement(int ID, EntityPlayer player, World world, int x, int y, int z){
        TileEntity tile = world.getTileEntity(x, y, z);
        switch(ID)
        {
            case 0:{
                return new GuiToolbox(new ContainerToolbox(((TileEntityToolbox) tile), player));
            }
            case 1:{
                return new GuiModifier(new ContainerModifier(((TileEntityModifier) tile), player));
            }
            case 2:{
                return new GuiTest(player, ((TileEntityModifier) tile), new TreeRecon());
            }
            case 3:{
                return new GuiDebug(player);
            }
            default:{
                return null;
            }
        }
    }
}